==== triton/lib/Dialect/Triton/IR/Ops.cpp - triton/lib/Dialect/Triton/IR/Ops.cpp ====
# action=edit type=text
--- triton/lib/Dialect/Triton/IR/Ops.cpp	2024-05-28 17:58:50.000000000 -0700
+++ triton/lib/Dialect/Triton/IR/Ops.cpp	2024-05-31 04:42:46.000000000 -0700
@@ -543,6 +543,8 @@
   auto value = adaptor.getSrc();
   if (!value)
     return {};
+  if (!isa<FloatAttr, IntegerAttr>(value))
+    return {};
   auto shapedType = cast<ShapedType>(getType());
   auto ret = SplatElementsAttr::get(shapedType, ArrayRef<Attribute>(value));
   return ret;
